### ISSUE TITLE
Issue with authorization code flow: Authorization code has expired

### ISSUE THREAD

**Issue by MikolayKorniat** [thephpleague/oauth2-server#1311]
I found a bug during the last switching from summer to winter time: [DateTime::add() adds wrong interval when switching from summer to winter time](https://bugs.php.net/bug.php?id=63311)

DateTimeImmutable::add() calculate wrong "[expire_time](https://github.com/thephpleague/oauth2-server/blob/0f32fbe00b6b1bdd72c8a70cb1763be49ce8c99b/src/Grant/AuthCodeGrant.php#L362
)" and issue already expired code.

Problem is fixed in PHP 8.1, but it's still actual for PHP =< 8.0. 

How about editing the timestamp directly instead of using a DateTimeImmutable?

**Comment by Sephster** [comment]
oh wow. Nice find. I had no idea this was an issue.

I'm leaning towards not changing this because we'd need a new major release anyways, it has been fixed in 8.1, and active support for 8.0 is ending in 11 days anyways.

That coupled with the fact that this bug manifests itself in a small time window makes me think it isn't worth the effort. Would appreciate your thoughts on this though.

I've reached out on Twitter to see if a patch will be issued for earlier versions but might not get a reply.
